MaterialLib/MPL/Properties/CapillaryPressureSaturation/CapillaryPressureVanGenuchten.h

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * \copydoc SaturationVanGenuchten
+ *
+ *  This class handles the computation of \f$ p_c(S) \f$ with the capillary
+ *  pressure regularization.
+ *
+ *   \brief The regularized van Genuchten model, please ref to
+ *   Marchand E, Mueller T, Knabner P.
+ *   Fully coupled generalized hybrid-mixed finite element approximation of
+ *   two-phase two-component flow in porous media.
+ *   Part I: formulation and properties of the mathematical model. Comput Geosci
+ *   2013;17(2):431-442.
+ */
+class RegularizedCapillaryVanGenuchten final
+    : public CapillaryPressureVanGenuchten
+{
+public:
+    RegularizedCapillaryVanGenuchten(double const residual_liquid_saturation,
+                                     double const residual_gas_saturation,
+                                     double const exponent,
+                                     double const entry_pressure,
+                                     double const max_capillary_pressure)
+        : CapillaryPressureVanGenuchten(residual_liquid_saturation,
+                                        residual_gas_saturation, exponent,
+                                        entry_pressure, max_capillary_pressure),
+          _saturation_nonwet_r(residual_gas_saturation)
+    {
+    }
+
+    void setScale(
+        std::variant<Medium*, Phase*, Component*> scale_pointer) override
+    {
+        if (!std::holds_alternative<Medium*>(scale_pointer))
+        {
+            OGS_FATAL(
+                "The property 'RegularizedCapillaryVanGenuchten' is "
+                "implemented on the "
+                "'media' scale only.");
+        }
+        this->_medium = std::get<Medium*>(scale_pointer);
+    }
+
+    /// gets \f$ p_c(S) \f$.
+    PropertyDataType value(VariableArray const& variable_array,
+                           ParameterLib::SpatialPosition const& pos,
+                           double const t,
+                           double const dt) const override;
+
+    /// gets \f$ \frac{\partial p_c(S)}{\partial  S} \f$
+    PropertyDataType dValue(VariableArray const& variable_array,
+                            Variable const variable,
+                            ParameterLib::SpatialPosition const& pos,
+                            double const t,
+                            double const dt) const override;
+
+    /// gets \f$ \frac{\partial^2 p_c(S)}{\partial  S^2} \f$
+    PropertyDataType d2Value(VariableArray const& variable_array,
+                             Variable const variable1, Variable const variable2,
+                             ParameterLib::SpatialPosition const& pos,
+                             double const t, double const dt) const override;
+
+private:
+    const double _saturation_nonwet_r;  ///< Residual saturation of nonwetting
+    static constexpr double _xi =
+        1e-5;  ///< parameter in regularized van Genuchten model
+
+    /// Regularized van Genuchten capillary pressure-saturation Model
+    double getPcBarvGSg(double const Sg) const;
```

Comment:
Comment does not match the function. A comment should describe a what a function does.

Reply:
Improved the comments.